### ISSUE TITLE
[BUG] Delay in showing RT/TMDB information

### ISSUE THREAD

**Issue by chcore** [jurialmunkey/skin.arctic.horizon.2#235]
### Current Behavior

When scrolling through the library, ratings other than the star (IMDB?) ratings are not rendered at the same time as the rest of the metadata, but only after a short delay. This causes a bit of a re-shuffle effect that looks a little ugly. Is this intended?

### Expected Behavior

The RT/TMDB ratings metadata should be updated instantly, like the star ratings or any other metadata related to the selected title.

### Steps To Reproduce

Scroll through your movies library and notice the delay in the display of ratings metadata.

### Checklist

- [X] I have searched the issues and this bug has not been reported.
- [X] I am using the most recent skin version.
- [X] I have confirmed that this bug does not occur in the default skin.

**Comment by jurialmunkey** [comment]
Impossible to fix. 

Ratings data for ListItems is not accessible outside the info dialog so the TMDbHelper service monitor is forced to look these up on the fly for the focused item.

The delay in the lookup is for two main reasons: 
- The ratings details actually need to be looked up!
  - Online lookup = Slowest (new metadata looked up)
  - Disk Cache lookup = Faster (metadata retrieved in a previous Kodi session)
  - Memory lookup = Fastest (metadata retrieved in the current Kodi session)
- ListItem properties cannot be modified after creation, so TMDbH must set the new metadata to the Window instead.
  - There is only one Window, so only properties for one item can be set at a time
  - As a result, all the Window properties must first be cleared and then reset on every focus change.
  - TMDbH can only know what to look-up and set after the item gains focus.
  - There is a further delay while the GUI skin engine retrieves those newly set properties from the Window

If Kodi allowed ListItems to be modified after creation there wouldn't be as much of a problem because then TMDbH could retrieve the next item and set its properties before it gained focus.

**Comment by chcore** [comment]
Thank you @jurialmunkey for the detailed explanation.

Thinking of potential work-arounds, are either of these possible or desirable?

1. Fade the ratings elements in once they are loaded
2. Have placeholder values so that we don't see the ugly re-shuffle of elements, e.g. display an initial RT rating of "N/A", then update its value if/when it's loaded.

**Comment by jurialmunkey** [comment]
Hmm, I'm thinking about this fade approach. I guess the concern there is when the ratings *do* load quickly an animation will then *cause* a delay because it needs to complete. Also it pulsing in and out could possible be even more distracting. 

I can imagine what you're picturing with the fade animation though. 

The N/A is also a decent idea (and could be a decent option once #209 is done). At least then the elements will be static positions. 

The downside is we get a lot of N/As in some areas (e.g. RT data is missing on OMDb for a lot of TV Shows). Also, the N/A might be confusing and give users the impression that the show isn't popular enough to even have a rating (when the reality is that OMDb is just missing the data).

I'll have a bit of play around with both and see if either is feasible. Anything that can give more consistent positioning and avoid distracting movement on screen is a win in my book.

**Comment by jurialmunkey** [comment]
Fade In was no good. The pulsing makes things worse.

**Comment by chcore** [comment]
I suppose one option is making the ratings only show (fade in) once the title has been selected for say at least 1 second?

And for the other work-around, maybe have a minimum amount of padding that is greater than both "N/A" and the rating (say 3 digits)?

Just throwing out ideas, not expecting these to be amazing.

**Comment by jurialmunkey** [comment]
I did try a longer delay but still wasn't great. It looks okay but I quickly found it frustrating waiting for the info to update.

The N/A approach is more promising but fixed width spacing is not great as it leaves inconsistent gaps.

One thought I had just now is to use dashes instead so a `9.0` style rating becomes `-.-` and a `90%` becomes `--%` when updating. That should help space more consistent though I'll need t to try it out in action. It's basically a case of finding a character that has the same kerning as the numbers but also clearly indicates an empty field.

**Comment by chcore** [comment]
The delayed fade-in probably wouldn't bother me as much (though I'm only imagining it), but I think I would be happy with either of those over the current behaviour.

**Comment by jurialmunkey** [comment]
Yeah I had a bit of a play with the dashes approach last night and it works quite nicely. It will definitely need to wait for the option to disable/enable specific ratings because it is a bit weird to constantly see blanks for metacritic and rotten tomatoes in the tvshow library but it works really nicely when scrolling in movies.
